CMake/Version.cmake

```diff
@@ -0,0 +1,84 @@
+
+# Try to figure out  versioning scheme for dynamic libraries
+# based on the system we're compiling for
+function(determine_version_type _out_version_type_name)
+
+        set(_ver_type "none")
+
+        if(WIN32 OR CYGWIN)
+                set(_ver_type "windows")
```

Comment:
I think cygwin should be "cygwin" and also have the versioning. DLLs built for cygwin do use versioning afaics.

Reply:
You are probably right about versioning but the variable case I must check. I was positive it's caps.

Reply:
I don't know about the cmake symbol whether it's CYGWIN or cygwin but I'll guess you have it right. What I meant is it should be `set(_ver_type "cygwin")`

Reply:
Oh, I see. Yeah, I was following libtool naming but I guess it doesn't make much sense to strictly keep to it.
Will do.